appservice/src/createAppService/createWebApp.ts

```diff
@@ -23,28 +23,38 @@ export async function createWebApp(
 }
 
 export async function setWizardContextDefaults(wizardContext: IAppServiceWizardContext, actionContext: IActionContext, advancedCreation?: boolean): Promise<void> {
+    const isJavaProject: boolean = await javaUtils.isJavaProject();
+    if (isJavaProject) {
+        wizardContext.recommendedSiteRuntime = new Map([
+            [LinuxRuntimes.java, 1],
+            [LinuxRuntimes.tomcat, 1],
+            [LinuxRuntimes.wildfly, 1]
+        ]);
+    }
     // only detect if one workspace is opened
     if (workspace.workspaceFolders && workspace.workspaceFolders.length === 1) {
         const fsPath: string = workspace.workspaceFolders[0].uri.fsPath;
         if (await fse.pathExists(path.join(fsPath, 'package.json'))) {
-            wizardContext.recommendedSiteRuntime = LinuxRuntimes.node;
+            wizardContext.recommendedSiteRuntime = new Map([[LinuxRuntimes.node, 1]]);
         } else if (await fse.pathExists(path.join(fsPath, 'requirements.txt'))) {
             // requirements.txt are used to pip install so a good way to determine it's a Python app
-            wizardContext.recommendedSiteRuntime = LinuxRuntimes.python;
+            wizardContext.recommendedSiteRuntime = new Map([[LinuxRuntimes.python, 1]]);
         }
-        actionContext.properties.recommendedSiteRuntime = wizardContext.recommendedSiteRuntime;
+        actionContext.properties.recommendedSiteRuntime = wizardContext.recommendedSiteRuntime ? [...wizardContext.recommendedSiteRuntime.keys()].join(',') : undefined;
     }
 
     if (!advancedCreation) {
         await LocationListStep.setLocation(wizardContext, 'centralus');
+        if (isJavaProject) {
+            // considering high resource requirement for Java applications, a higher plan sku is set here
+            wizardContext.newPlanSku = { name: 'P1v2', tier: 'PremiumV2', size: 'P1v2', family: 'P', capacity: 1 };
+            // to avoid 'Requested features are not supported in region' error
+            await LocationListStep.setLocation(wizardContext, 'westeurope');
```

Comment:
I found the `P1v2` is not always available in `Central US`. I've met this problem, but not stable repro.

So the location is set to `West Europe` here.

Reply:
Sounds fine to me. I don't have a strong preference on default location.

Reply:
Ok. I'll revert the code for switching location once the issue has been addressed.